### PR TITLE
refetch queries on mutation not working properly

### PR DESCRIPTION
**Pull request by Lavanyagaur22**
### Description

When I am refetching the queries to update cache on performing any mutation, it's not working properly. When I clear data of the app, then only proper results are shown. Although, data in the cache is correct ( I have gone through the data ), but the queries are not showing the correct data. They are showing older data.

### CODE REVIEW

**Comment by wtrocki** [comment]
2 problems. Refetch queries is async and we do not connect UI on success, but even after that we are using default cache strategy that may nit be right for this case. Check FetchPolicy argument

File: sample/src/main/java/org/aerogear/graphqlandroid/activities/MainActivity.kt [hunk 1]
```diff
@@ -107,30 +134,39 @@ class MainActivity : AppCompatActivity() {
...
-            UpdateCurrentTask.builder().id(id).title(title).version(version).build()
-        )
-        client?.enqueue(object : ApolloCall.Callback<UpdateCurrentTask.Data>() {
-            override fun onFailure(e: ApolloException) {
-                Log.e("onFailure" + "updateTask", e.toString())
+        val client: ApolloMutationCall<UpdateCurrentTask.Data>?
+
+        val mutation = UpdateCurrentTask.builder().id(id).title(title).version(version).build()
+
+        client = Utils.getApolloClient(this)?.mutate(mutation)?.refetchQueries(object : OperationName {
```
Comment (harshithdwivedi) [inline sample/src/main/java/org/aerogear/graphqlandroid/activities/MainActivity.kt:141]: `refetchQueries()` requires an object of the type `OperationName` or `Query`.

What you've passed here looks like the `OperationDefinition`, why don't you get the `OPERATION_NAME` from `AllTaskQuery` and pass it here instead?

**Comment by Lavanyagaur22** [comment]
I am not able to access ApolloResponseFetchers,
 `import com.apollographql.apollo.fetcher.ApolloResponseFetchers` is not being imported

**Comment by Lavanyagaur22** [comment]
> 2 problems. Refetch queries is async and we do not connect UI on success, but even after that we are using default cache strategy that may nit be right for this case. Check FetchPolicy argument

Refetch queries which is async doesn't have a callback, how would we get to know that it has succeeded or not?

**Comment by wtrocki** [comment]
> Refetch queries which are async doesn't have a callback, how would we get to know that it has succeeded or not?

Refetch queries is evil. What problem we are trying to resolve with refetch queries?

**Pull request closed without merging**